tests/test_lnpeer.py

```diff
@@ -1970,6 +1970,71 @@ async def f():
         with self.assertRaises(SuccessfulTest):
             await f()
 
+    async def test_dont_settle_htlcs(self):
+        """
+        Test that htlcs registered in LNWallet.dont_settle_htlcs don't get fulfilled if the preimage is available.
+        """
+        async def run_test(test_trampoline, test_failure):
+            alice_channel, bob_channel = create_test_channels()
+            p1, p2, w1, w2, _q1, _q2 = self.prepare_peers(alice_channel, bob_channel)
+            if test_trampoline:
+                await self._activate_trampoline(w1)
+                # declare bob as trampoline node
+                electrum.trampoline._TRAMPOLINE_NODES_UNITTESTS = {
+                    'bob': LNPeerAddr(host="127.0.0.1", port=9735, pubkey=w2.node_keypair.pubkey),
+                }
+
+            preimage = os.urandom(32)
+            lnaddr, pay_req = self.prepare_invoice(w2, payment_preimage=preimage, min_final_cltv_delta=244)
```

Comment:
why 244? why not the default?
is it because 20 lines below you mine a few blocks, and the HTLC would get failed due to expiry getting close?